### PR TITLE
Implement DB related requirements

### PR DESCRIPTION
**Pull request by EdvardasDlugauskas**
Also changed the DB to use GUIDs for IDs

### CODE REVIEW

**Comment by marjasi** [comment]
[[Requirement] Select/insert/update/delete.](https://trello.com/c/5tDxMVVD/94-requirement-select-insert-update-delete)

**Comment by marjasi** [comment]
[[Requirement] LINQ.](https://trello.com/c/gKGDw8lz/95-requirement-linq)

**Comment by marjasi** [comment]
[[Requirement] DataTable, DataAdapter usage.](https://trello.com/c/adPb2qEq/93-requirement-datatable-dataadapter-usage)

File: Mobile/ShopLens/MigrationsConsole/Program.cs [hunk 1]
```diff
@@ -58,11 +74,26 @@ public static void ListAllUsers(ShopLensContext dbContext)
                 Console.WriteLine("No users found.");
                 return;
             }
-
-            foreach (var user in users)
+            
+            foreach(var user in users)
             {
-                Console.WriteLine($"User {user.UserId}: Name - {user.Name}, BDay - {user.Birthday}");
+                Console.WriteLine($"User {user.UserId}: Name - {user.Name}, BDay - {user.Birthday}") ;
```
Comment (marjasi) [inline Mobile/ShopLens/MigrationsConsole/Program.cs]: I see that you added a space here.

File: Mobile/ShopLens/MigrationsConsole/Requirements/SqlRequirements.cs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            Console.WriteLine("(.Join) Users and their shopping sessions:");
+            foreach (var userAndShoppingSession in usersAndShoppingSessions)
+            {
+                Console.WriteLine(userAndShoppingSession);
+            }
+        }
+
+        ~SqlRequirements()
+        {
+            connection.Close();
```
Comment (marjasi) [inline Mobile/ShopLens/MigrationsConsole/Requirements/SqlRequirements.cs]: The finalizer throws an exception when I tested it.
image

**Review by marjasi — CHANGES_REQUESTED** [review]
Hey, good job 😄 👍 , I'm so relieved that we have the requirements done already. Despite that, I get an exception error, so I'll request changes.

File: Mobile/ShopLens/MigrationsConsole/Requirements/SqlRequirements.cs (hunk 2 above)
Comment (EdvardasDlugauskas) [reply Mobile/ShopLens/MigrationsConsole/Requirements/SqlRequirements.cs]: Could you perhaps try running it once or twice more? Try commenting out some of the statements like lines 36-40 in Program.cs as well.

It's weird, I got it to crash once, then commented all statements out, it went fine, then uncommented the statements (original code) and now it executes fine again (???) I'm not sure what's going on

Comment (marjasi) [reply Mobile/ShopLens/MigrationsConsole/Requirements/SqlRequirements.cs]: I commented lines 36-40, like you said, then I commented the finalizer of the Requirements class and I got this error.
image
This was on a local DB, so I switch over to our Azure DB. Then everything was working with with the lines 36-40 and the finalizer commented out.
image
I uncommented lines 36-40 and on our Azure DB it worker like a charm. 
image
Then I uncommented the finalizer and I got the error again.
image
I tried it again several times on a local DB with the finalizer commented out and I got the Users error, but no Driver error. 
image
Then I tried it several times with the finalizer commented out on our Azure DB and it worked okay.

Comment (marjasi) [reply Mobile/ShopLens/MigrationsConsole/Requirements/SqlRequirements.cs]: Houston, I think we have found the problem. 
image

Comment (marjasi) [reply Mobile/ShopLens/MigrationsConsole/Requirements/SqlRequirements.cs]: Still, I didn't know anything about finalizers, so thanks for showing us that they exist.

**Review by marjasi — APPROVED** [review]

**Pull request merged**